### PR TITLE
Mention linking exception for GPLed libiberty, cleanup

### PR DESCRIPTION
**Pull request by StefanBruens**
The C++ demangling code from libiberty is GPL-2.0 or later, but has a linking
exception.

Remove no longer existing gnu arch reference.

**Checklist**

- [ ] Closing issues: #issue
- [x] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**

### CODE REVIEW

File: doc/license.md [hunk 1]
```diff
@@ -77,8 +77,7 @@ should be copied to `./plugins.cfg` before calling `./configure-plugins` to take
 
 ### Less restrictive than LGPL
 
-* libr/asm/arch/gnu: GPLv2
```
Comment (trufae) [inline doc/license.md:80]: Dont remove this line

Comment (trufae) [reply doc/license.md:80]: The code in this subdirectory is from gnu binutils

Comment (StefanBruens) [reply doc/license.md:80]: This subdirectory does not even exist.

Comment (trufae) [reply doc/license.md:80]: Correct. Its libr/asm/arch/*/gnu

File: doc/license.md [hunk 2]
```diff
@@ -77,8 +77,7 @@ should be copied to `./plugins.cfg` before calling `./configure-plugins` to take
...
-* libr/bin/mangling/cxx: GPLv2
+* libr/bin/mangling/cxx: GPLv2 or later, with linking exception
```
Comment (radare) [inline doc/license.md:80]:
```suggestion
* libr/asm/arch/*/gnu: GPLv2
* libr/bin/mangling/cxx: GPLv2 or later, with linking exception
```

**Review by radare — CHANGES_REQUESTED** [review]
.

**Comment by radare** [comment]
Please rebase to make it green

File: doc/license.md (hunk 2 above)
Comment (StefanBruens) [reply doc/license.md:80]: GPLv2 is obviously *not* "Less restrictive than LGPL". The code in */gnu has no linking exception, AFAICS.

Comment (trufae) [reply doc/license.md:80]: Correct, no linking exception in binutils

**Pull request merged**